Sources/NIOConcurrencyHelpers/NIOLockedValueBox.swift

```diff
@@ -37,6 +37,43 @@ public struct NIOLockedValueBox<Value> {
     public func withLockedValue<T>(_ mutate: (inout Value) throws -> T) rethrows -> T {
         return try self._storage.withLockedValue(mutate)
     }
+
+    /// Provides an unsafe view over the lock and its value.
+    ///
+    /// This can be beneficial when you require fine grained control over the lock in some
+    /// situations but don't want lose the benefits of ``withLockedValue(_:)`` in others by
+    /// switching to ``NIOLock``.
+    public var unsafe: Unsafe {
+        Unsafe(_storage: self._storage)
+    }
+
+    /// Provides an unsafe view over the lock and its value.
+    public struct Unsafe {
+        @usableFromInline
+        let _storage: LockStorage<Value>
+
+        /// Manually acquire the lock.
+        @inlinable
+        public func lock() {
+            self._storage.lock()
+        }
+
+        /// Manually release the lock.
+        @inlinable
+        public func unlock() {
+            self._storage.unlock()
+        }
+
+        /// Mutate the value, assuming the lock has been acquired manually.
+        @inlinable
+        public func withValueAssumingLockIsAcquired<T>(
```

Comment:
The spelling for this on `OSAllocatedUnfairLock` is `withLockUnchecked`. It might be worth us aligning with that.

While we're here, let's avoid using single-letter generic type parameter names. `ReturnValue` is  probably better.

Reply:
Do you think `Unchecked` is also a more appropriate name for the type as well? Wasn't so sure on the naming...

Reply:
I think `Unchecked` is better. We normally use `Unsafe` when it causes undefined behaviour which manual locking shouldn't

Reply:
IIUC `withLockUnchecked` has different semantics: it still acquires the lock but does not check the sendability of the closure passed in.

This is different: it assumes that the caller has already called `lock()` and that they will subsequently call `unlock()`.

Reply:
re: `Unsafe`, I think depending on what is stored in the locks state, this _could_ be unsafe if you didn't acquire the lock first.

Reply:
Ok, let's leave the name and fix the generic type parameters.

And FWIW, getting manual locking wrong can absolutely cause UB.